compiler/src/middle_end/linearize.re

```diff
@@ -1329,25 +1329,42 @@ let rec transl_anf_statement =
     };
   | TTopException(_, ext) => (Some(linearize_exception(env, ext)), [])
   | TTopForeign(exported, desc) =>
-    let (argsty, retty) =
-      get_fn_allocation_type(env, desc.tvd_desc.ctyp_type);
     let global =
       switch (exported) {
       | Exported => Global
       | Nonexported => Nonglobal
       };
-    (
-      None,
-      [
-        Imp.wasm_func(
-          ~global,
-          desc.tvd_id,
-          desc.tvd_mod.txt,
-          desc.tvd_name.txt,
-          FunctionShape(argsty, [retty]),
-        ),
-      ],
-    );
+    switch (desc.tvd_desc.ctyp_type.desc) {
```

Comment:
Heh yeah, it does look pretty crazy.

Reply:
Can you explain it? I kinda know `ctyp` but I've never seen `tvd`

Reply:
In plain words, this gets the type info out of a `value_description`.

Longer:
`desc` here is a `value_description` from `typedtree.re`, which describes all values that exist in the environment, with info about what module it came from, its type, whether or not it's a primitive, etc. and `tvd_desc` is the core type of that value. (it could probably be called `tvd_type`, but we co-opted the name from OCaml. I think it's called that because a core type is a sort of description of a type). `cty_type` is the type expression from that core type.